appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/node/DeploymentDescriptorNode.java

```diff
@@ -1120,15 +1120,16 @@ public void appendQNameChild(String elementName, Node parent, String namespaceUr
 
         String elementValue = composeQNameValue(prefix, localPart);
         Element element = (Element) appendTextChild(parent, elementName, elementValue);
-
         // Always set prefix mapping on leaf node. If the DOL was
         // populated from an existing deployment descriptor it does
         // not preserve the original node structure of the XML document,
         // so we can't reliably know what level to place mapping.
         // Alternatively, if we're writing out a descriptor that was created
         // by the deploytool, there is no prefix->namespace information in
         // the first place.
-        setAttributeNS(element, prefix, namespaceUri);
+        if (element != null) {
+            setAttributeNS(element, prefix, namespaceUri);
```

Comment:
Should the `element` be null (empty text) we also don't want to add the attribute. Behaviour change: NPE becomes NOOP and now allows adding empty text nodes.